examples/lie_labs.py

```diff
@@ -0,0 +1,121 @@
+import torch
+
+import theseus.labs.lie as lie
+import theseus.labs.lie.functional as lieF
+
+batch_size = 5
+
+# ### Lie Tensor creation functions
+g1 = lie.rand(batch_size, lie.SE3, requires_grad=True)
+print(f"Created SE3 tensor with shape {g1.shape}")
+g2 = g1.clone()
+
+# Can create from a tensor as long as it's consistent with the desired ltype
+g3_data = lieF.so3.rand(5)
+g3 = lie.from_tensor(g3_data, lie.SO3)
+
+try:
+    x = lie.from_tensor(torch.zeros(1, 3, 3), lie.SO3)
+except ValueError as e:
+    print(f"ERROR: {e}")
+
+g4 = lie.as_lietensor(g3_data, lie.SO3)
+g5 = lie.cast(g3_data, lie.SO3)  # alias for as_lietensor
+
+# ### Lie operations
+v = torch.randn(batch_size, 6)
+
+# Exponential and logarithmic map
+out1 = g1.exp(v)  # also lie.exp(v, g1.ltype)
+print(f"Exp map returns a {type(out1)}.")
+out2 = g1.log()  # also lie.log(g1)
+print(f"Log map returns a {type(out2)}.")
+
+# Inverse
+out1 = g1.inv()  # also lie.inv(g1)
+
+# Compose
+# also lie.compose(g1, g2)
+out1 = g1.compose(g2)  # type: ignore
+
+# Differentiable jacobians
+jacs, out = g1.jcompose(g2)  # type: ignore
+print("Jacobians output is a 2-tuple.")
+print("    First element is a list of jacobians, one per group argument.")
+print(f"    For compose this means length {len(jacs)}.")
+print("    The second element of the tuple is the result of the operation itself.")
+print(f"    Which for compose is a {type(out).__name__}.")
+
+# Other options:
+#   * adj(), hat(), vee(), retract(), local(),
+#   * Jacobians: jlog(), jinv(), jexp()
+
+# ### Overriden operators
+# Local map
+z = g2 - g1  # equivalent to g1.local(g2)
+print(f"Output of local is a {type(z)}.")
+g4 = -g1  # equivalent to g1.inv()
+torch.testing.assert_close(g1.inv(), g4)
+
+# Some operators have strict typecheck requirements
+try:
+    g2 - z
+except TypeError as e:
+    print(f"ERROR: {e}")
+
+try:
+    g3 - g1
+except ValueError as e:
+    print(f"ERROR: {e}")
+
+# For retract(), which can be written as g + x, the requirement is that x
+# is a LieTensor of ltype=lie.tgt, which we require as an extra layer of safety.
+# 6 is the DOF of SE3
+x = torch.randn(batch_size, 6)  # type: ignore
+try:
+    g1 + x
+except TypeError as e:
+    print(f"ERROR: {e}")
+
+y = lie.cast(x, lie.tgt)
+g2 = g1 + y
```

Comment:
I find it slightly inconsistant that the operations such as SE3.exp(v) allows v to be a tensor, where as the overloaded manifold plus which is g1.compose(SE3.exp(v)) doesn't allow v to be a tensor and you need the cast it to lie.tgt

To be consistent we should either allow it or not allow it for both? It's probably going to be weird writing code where in some part this cast is required and in others it's not

Reply:
One concern we have had since the beginning is people trying to do confusing stuff with overloaded operators and have an operation going through with a calculation that makes absolutely no sense. For example, some naive user wanting to do standard addition on the data matrix, and just passing `g + x` without realizing this is actually a retraction map. That's the kind of silent error we are trying to avoid with this. It's different with `SE3.retract()` because there is no ambiguity about what they are trying to do. 

Do you think this concern is overstated? Any thoughts from @exhaustin, @dongjing3309, @ddetone?

Reply:
Disclaimer: This is my own personal preference and might not apply to some or the majority of users.
I would be fine without operator overloading, as it makes the code look more transparent and easier to debug.
The one exception is `compose` as it is extremely common and relatively straightforward. Overloading the `@` operator with that functionality is nice as 1) it is already a common operator for matrix multiplications, and performing matrix operations on the data matrix is AFAIU equivalent to compose? 2) It won't be mistaken as an elementwise multiplication. (Personally I'm not against `*` either)

Reply:
Thanks for sharing! This is valuable input. Would love to hear from others, but FWIW, I think this aligns well with @fantaosha's opinion (he is concerned about the transparency/ambiguity of these operators).

Reply:
Removed `+` in 530172f5005a92b0eb52446092cd3f8620c673d9.